extensions/amp-bind/amp-bind.md

```diff
@@ -37,66 +37,80 @@ limitations under the License.
+3. The &lt;p&gt;'s text changes to show "Hello World".
 
-<p [text]=”foo.message”>Placeholder text</p>
+### Initializing state
+
+The implicit state can be initialized at page load with custom JSON. For example:
+
+```html
+<amp-state id="myState">
+  <script type="application/json">
+    {"foo": "bar"}
+  </script>
+</amp-state>
 ```
 
-1. `amp-bind` scans the DOM and finds the `<p>` element’s `[text]` binding.
-2. During the next digest, `amp-bind` reevaluates the expression `foo.message`.
-3. On the next frame, `amp-bind` updates the `<p>` element's textContent from "Placeholder text" to "Hello World".
+Binding expressions can reference the data in this `<amp-state>` component via dot syntax.
+In this example, `myState.foo` will evaluate to `"bar"`.
 
-## Binding
+## Data binding
 
-`amp-bind` supports binding to three types of data on an element:
+`amp-bind` supports data bindings on three types of element state:
 
 | Type | Syntax | Details |
 | --- | --- | --- |
-| Attribute | `[<attr>]` | Only whitelisted attributes are supported.<br>Boolean expression results toggle boolean attributes.
-| CSS Classes | `[class]` | Expression result must be a space-delimited string.
-| Node.textContent | `[text]` | For applicable elements.
-
-**Caveats:**
+| [Node.textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) | `[text]` | Supported on most text elements.
+| [CSS Classes](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/class) | `[class]` | Expression result must be a space-delimited string.
+| Element-specific attribute | `[<attr>]` | Only whitelisted attributes are supported.<br>Boolean expression results toggle boolean attributes.
 
 - For security reasons, binding to `innerHTML` is disallowed
 - All attribute bindings are sanitized for unsafe URL protocols (e.g. `javascript:`)
 
 ### Bindable attributes
 
-For non-AMP elements, generally all attributes are bindable.
+For non-AMP elements, most attributes accepted by the [AMP Validator](https://validator.ampproject.org/) are bindable.
 
-For AMP components, only a subset of attributes are bindable:
+For AMP components, only select attributes are bindable:
 
 | Component | Attributes |
 | --- | --- |
-| amp-img | src |
-| amp-video | src |
-| amp-pixel | src |
+| amp-carousel | slide |
+| amp-img | src, srcset, alt |
+| amp-selector | selected |
+| amp-video | src, srcset, alt, controls, loop, poster |
 
-Note that the set of bindable attributes will grow over time as development progresses.
+See [BindValidator](./0.1/bind-validator.js) for the canonical set of bindable
+elements and attributes.
 
 ## Expressions
 
 `amp-bind` expressions are JS-like with some important differences:
 
-- Expressions only have access to the **scope**, not globals like `window` or `document`
+- Expressions may only access JSON state in `<amp-state>` and that set by the `AMP.setState` action
```

Comment:
This sentence is slightly confusing because of the "and that set by" - *what* is set by the action? Is it the expression or the state? I think it's the state, so consider rewriting it as follows:

Expressions may only access the JSON state specified in `<amp-state>`, and that state is set by the `AMP.setState` action.